### PR TITLE
Milestone 8:- Add views specs

### PR DESCRIPTION
**Pull request by hmunish**
Tasks completed:-

- Fixed N+1 problems for controllers
- Used capybara for testing each view
- Fixed linter errors

user-control-old
user-control-new
post-control-old
post-control-new

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
+    @user = @user = User.includes(posts: :comments).find(params[:user_id])
     @posts = @user.posts.includes(:comments)
```
Comment (TadesseDev) [inline app/controllers/posts_controller.rb:4]: - [ ] Dear Team, It is crucial that you add In your PR description the screenshots of your app's logs before and after the fix while dealing with N+1. [Refer here for requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=Do%20not%20use,after%20the%20fix).

File: spec/features/user_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      @user1 = User.create(
+        name: 'Munish',
+        photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+        bio: 'Web developer from Malawi',
+        posts_counter: 0
+      ),
+      @user2 = User.create(
+        name: 'Halai',
+        photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+        bio: 'Software Engineer.',
```
Comment (TadesseDev) [inline spec/features/user_index_spec.rb:15]: - [ ] Dear @hmunish, For the profile picture we need to make sure we `can see the profile picture for each user`, hence Please apply a different profile picture and check if both are visible. Currently even if one of the profile picture is not visible the test will pass, and this is not the expected behaviour.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  All tests are inplce✅
 - [x]  Most of the requirement has been meet. ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @hmunish and @Othniel2471,

## Highlights!
- Good code organization and tests ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**